### PR TITLE
Updated lwan-array.h for c++ compatibility

### PR DESCRIPTION
**Pull request by GuacheSuede**

### CODE REVIEW

File: common/lwan-array.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ struct lwan_array *coro_lwan_array_new(struct coro *coro);
     } \
     static inline element_type_ * array_type_ ## _append(struct array_type_ *array) \
     { \
-        return lwan_array_append((struct lwan_array *)array, sizeof(element_type_)); \
+        return (struct lwan_key_value *)lwan_array_append((struct lwan_array *)array, sizeof(element_type_)); \
```
Comment (lpereira) [inline common/lwan-array.h]: This should be `(element_type_ *)` instead.

**Comment by GuacheSuede** [comment]
@lpereira Quick offtopic, how do I manually set headers in lwan ?

File: common/lwan-array.h (hunk 1 above)
Comment (GuacheSuede) [inline common/lwan-array.h]: Added further change

File: common/lwan-array.h [hunk 2]
```diff
@@ -40,19 +40,19 @@ struct lwan_array *coro_lwan_array_new(struct coro *coro);
     }; \
     static inline int array_type_ ## _init(struct array_type_ *array) \
     { \
-        return lwan_array_init((struct lwan_array *)array); \
+        return (int)lwan_array_init((struct lwan_array *)array); \
```
Comment (lpereira) [inline common/lwan-array.h]: Why?

File: common/lwan-array.h [hunk 3]
```diff
@@ -40,19 +40,19 @@ struct lwan_array *coro_lwan_array_new(struct coro *coro);
...
     } \
     static inline element_type_ * array_type_ ## _append(struct array_type_ *array) \
     { \
-        return (struct lwan_key_value *)lwan_array_append((struct lwan_array *)array, sizeof(element_type_)); \
+        return (element_type_ *)lwan_array_append((struct lwan_array *)array, sizeof(element_type_)); \
     } \
     static inline void array_type_ ## _sort(struct array_type_ *array, int (*cmp)(const void *a, const void *b)) \
     { \
-        lwan_array_sort((struct lwan_array *)array, sizeof(element_type_), cmp); \
+        (void)lwan_array_sort((struct lwan_array *)array, sizeof(element_type_), cmp); \
```
Comment (lpereira) [inline common/lwan-array.h]: Why?

File: common/lwan-array.h [hunk 4]
```diff
@@ -40,19 +40,19 @@ struct lwan_array *coro_lwan_array_new(struct coro *coro);
     }; \
     static inline int array_type_ ## _init(struct array_type_ *array) \
     { \
-        return lwan_array_init((struct lwan_array *)array); \
+        return (int)lwan_array_init((struct lwan_array *)array); \
     } \
     static inline int array_type_ ## _reset(struct array_type_ *array) \
     { \
-        return lwan_array_reset((struct lwan_array *)array); \
+        return (int)lwan_array_reset((struct lwan_array *)array); \
```
Comment (lpereira) [inline common/lwan-array.h]: Why?

**Comment by lpereira** [comment]
@GuacheSuede You can take a look in the testrunner/main.c, there are a few tests there that do that.

File: common/lwan-array.h (hunk 2 above)
Comment (GuacheSuede) [inline common/lwan-array.h]: Just felt appropriate, will remove, seems uneeded.

Comment (lpereira) [inline common/lwan-array.h]: Yeah, lwan_array_init() already returns an int.

**Comment by lpereira** [comment]
@GuacheSuede Please squash all of these changes in a single commit. [Here's a guide](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).

**Comment by GuacheSuede** [comment]
Alright, do take a look at the updated squash

**Comment by lpereira** [comment]
Merged, thanks!

**Pull request merged**